libs/perun/dialogs/src/lib/password-reset-dialog/password-reset-dialog.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import {
+  ApiRequestConfigurationService,
+  NotificatorService,
+  StoreService,
+} from '@perun-web-apps/perun/services';
+import { UsersManagerService } from '@perun-web-apps/perun/openapi';
+import { loginAsyncValidator } from '@perun-web-apps/perun/namespace-password-form';
+import { FormBuilder, FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
+import { CustomValidators } from '@perun-web-apps/perun/utils';
+import { PasswordLabels } from '@perun-web-apps/perun/models';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+
+export interface ResetPasswordDialogData {
+  namespace: string;
+  login: string;
+}
+
+@Component({
+  selector: 'perun-web-apps-password-reset-dialog',
+  templateUrl: './password-reset-dialog.component.html',
+  styleUrls: ['./password-reset-dialog.component.scss'],
+})
+export class PasswordResetDialogComponent implements OnInit {
+  loading = false;
+  success = false;
+  error = false;
+  language = 'en';
+  newPasswdForm: FormGroup<{
+    passwordCtrl: FormControl<string>;
+    passwordAgainCtrl: FormControl<string>;
+  }>;
+  labels: PasswordLabels;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: ResetPasswordDialogData,
+    private dialogRef: MatDialogRef<PasswordResetDialogComponent>,
+    private storeService: StoreService,
+    private translate: TranslateService,
+    private apiRequestConfiguration: ApiRequestConfigurationService,
+    private usersService: UsersManagerService,
+    private formBuilder: FormBuilder,
+    private notificator: NotificatorService
+  ) {}
+
+  ngOnInit(): void {
+    this.newPasswdForm = this.formBuilder.group(
+      {
+        passwordCtrl: [
+          '',
+          Validators.required,
+          [
+            loginAsyncValidator(
+              this.data.namespace,
+              this.usersService,
+              this.apiRequestConfiguration
+            ),
+          ],
+        ],
+        passwordAgainCtrl: ['', Validators.required],
+      },
+      {
+        validators: CustomValidators.passwordMatchValidator as ValidatorFn,
+      }
+    );
+    this.setLabels(this.translate.currentLang);
+  }
+
+  onSubmit(): void {
+    this.loading = true;
+    this.usersService
+      .changePasswordForLogin({
+        login: this.data.login,
+        namespace: this.data.namespace,
+        newPassword: this.newPasswdForm.value.passwordCtrl,
+      })
+      .subscribe(() => {
+        this.success = true;
+        this.notificator.showSuccess(
+          this.translate.instant(
+            'SHARED_LIB.PERUN.COMPONENTS.RESET_PASSWORD_DIALOG.SUCCESS'
+          ) as string
+        );
```

Comment:
This method takes care of the translation itself and is the preferred way to show notifications.
```suggestion
        this.notificator.showInstantSuccess(
          'SHARED_LIB.PERUN.COMPONENTS.RESET_PASSWORD_DIALOG.SUCCESS'
        );
```